cmd/commands/image.go

```diff
@@ -0,0 +1,99 @@
+		Use:   "image",
+		Short: "Interact with docker images",
+	}
+}
+
+func ImageRelocate(c *core.Client) *cobra.Command {
+	options := core.RelocateImageOptions{}
+
+	command := &cobra.Command{
+		Use:   "relocate",
+		Short: "Relocate docker image names to another registry",
+		Long: `Relocate either a single YAML file or a riff manifest and its YAML files so that image names refer to
+another (private or public) registry.
+
+To transform a single YAML file, use the '--yaml' flag to specify the path or URL of the YAML file. Use the '--output'
+flag to specify the path to contain the transformed YAML file. If '--output' is an existing directory, the YAML file
+will be created in that directory. Otherwise the YAML file will be written to the path specified in '--output'.
+
+To transform a manifest, use the '--manifest' flag to specify the path of a manifest file which provides the paths or
+URLs of the YAML definitions of riff components. Use the '--output' flag to specify the path of a directory to contain
+the transformed manifest and YAML definitions.
+
+Specify the registry hostname using the '--registry' flag, the user owning the images using the '--registry-user' flag,
+and a complete list of the images to be mapped using the '--images' flag. The '--images' flag contains the path of animage manifest file with contents of the following form:
+
+    manifestVersion: 0.1
+    images:
+    ...
+    - istio/sidecar_injector
+    ...
+    - gcr.io/knative-releases/github.com/knative/serving/cmd/autoscaler@sha256:76222399addc02454db9837ea3ff54bae29849168586051a9d0180daa2c1a805
+    ...
+    
+`,
+		Example: `  riff image relocate --manifest=/path/to/manifest --registry=hostname --user=username --images=/path/to/image/manifest --output=/path/to/output/dir
+  riff image relocate --yaml=/path/to/yaml/file --registry=hostname --user=username --images=/path/to/image/manifest --output=/path/to/output`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// TODO: implement support for global flags - for now don't allow their use
+			if cmd.Flags().Changed("kubeconfig") {
+				return errors.New("The 'kubeconfig' flag is not supported by the 'image relocate' command")
+			}
+			m, _ := cmd.Flags().GetString("master")
+			if len(m) > 0 {
+				return errors.New("The 'master' flag is not supported by the 'image relocate' command")
+			}
+			err := FlagsValidatorAsCobraRunE(
+				FlagsValidationConjunction(
+					ExactlyOneOf("yaml", "manifest"),
+					AllOf("registry", "registry-user", "images", "output"),
+				),
+			)(cmd, args)
+			return err
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := (*c).RelocateImage(options)
+			if err != nil {
+				return err
+			}
+
+			printSuccessfulCompletion(cmd)
+			return nil
+		},
+	}
+
+	command.Flags().StringVarP(&options.YAML, "yaml", "y", "", "file path of a YAML file")
+	command.Flags().StringVarP(&options.Manifest, "manifest", "m", "", "file path of a riff manifest file")
```

Comment:
In those sentences, does the word "file" assume it's gonna be a file (as opposed to a dir)? Because it's not true for `output`. I would drop it everywhere

Reply:
Yes. Fixed.